pkg/server/backend_manager.go

```diff
@@ -105,67 +132,93 @@ type DefaultBackendStorage struct {
 	// Golang.
 	agentIDs []string
 	random   *rand.Rand
+	// idTypes contains the valid identifier types for this
+	// DefaultBackendStorage. The DefaultBackendStorage may only tolerate certain
+	// types of identifiers when associating to a specific BackendManager,
+	// e.g., when associating to the DestHostBackendManager, it can only use the
+	// identifiers of types, IPv4, IPv6 and Host.
+	idTypes []pkgagent.IdentifierType
 }
 
 // NewDefaultBackendManager returns a DefaultBackendManager.
 func NewDefaultBackendManager() *DefaultBackendManager {
-	return &DefaultBackendManager{DefaultBackendStorage: NewDefaultBackendStorage()}
+	return &DefaultBackendManager{
+		DefaultBackendStorage: NewDefaultBackendStorage(
+			[]pkgagent.IdentifierType{pkgagent.UID})}
 }
 
 // NewDefaultBackendStorage returns a DefaultBackendStorage
-func NewDefaultBackendStorage() *DefaultBackendStorage {
+func NewDefaultBackendStorage(idTypes []pkgagent.IdentifierType) *DefaultBackendStorage {
 	return &DefaultBackendStorage{
 		backends: make(map[string][]*backend),
 		random:   rand.New(rand.NewSource(time.Now().UnixNano())),
+		idTypes:  idTypes,
+	}
+}
+
+func containIdType(idTypes []pkgagent.IdentifierType, idType pkgagent.IdentifierType) bool {
+	for _, it := range idTypes {
+		if it == idType {
+			return true
+		}
 	}
+	return false
 }
 
 // AddBackend adds a backend.
-func (s *DefaultBackendStorage) AddBackend(agentID string, conn agent.AgentService_ConnectServer) Backend {
+func (s *DefaultBackendStorage) AddBackend(identifier string, idType pkgagent.IdentifierType, conn agent.AgentService_ConnectServer) Backend {
+	if !containIdType(s.idTypes, idType) {
+		klog.ErrorS(&ErrWrongIDType{idType, s.idTypes}, "fail to add backend")
+		return nil
+	}
 	klog.V(2).InfoS("Register backend for agent", "connection", conn, "agentID", agentID)
 	s.mu.Lock()
 	defer s.mu.Unlock()
-	_, ok := s.backends[agentID]
+	_, ok := s.backends[identifier]
```

Comment:
At this point we have lost the id type, which means there is at least theoretically the possibility of a collision across types. Good news is we can always fix this later.

Reply:
will try to fix this issue in the next PR.